frontend/src/app/scripts/_components/ScriptInfoBlocks.tsx

```diff
@@ -121,29 +136,13 @@ export function LatestScripts({ items }: { items: Category[] }) {
 }
 
 export function MostViewedScripts({ items }: { items: Category[] }) {
-  const [page, setPage] = useState(1);
-
-  const mostViewedScripts = useMemo(() => {
-    if (!items) return [];
-    const scripts = items.flatMap((category) => category.expand.items || []);
-    const mostViewedScripts = scripts
-      .filter((script) => script.isMostViewed)
-      .map((script) => ({
-        ...script,
-      }));
-    return mostViewedScripts;
-  }, [items]);
-
-  const goToNextPage = () => {
-    setPage((prevPage) => prevPage + 1);
-  };
-
-  const goToPreviousPage = () => {
-    setPage((prevPage) => prevPage - 1);
-  };
-
-  const startIndex = (page - 1) * ITEMS_PER_PAGE;
-  const endIndex = page * ITEMS_PER_PAGE;
+  const mostViewedScripts = items.reduce((acc, category) => {
+    const foundScripts = category.scripts.filter((script) =>
+      mostPopularScripts.includes(script.name),
+    );
+    acc.push(...foundScripts);
+    return acc;
+  }, [] as Script[]);
```

Comment:
```suggestion
  }, []);
```
Do we need to force type here, does typescript not annotate it correctly?

Reply:
yeah, for some reason it doesnt work without it

Reply:
Maybe just `const mostViewedScripts: Script[] = ` instead then?